src/lib/molecules/Charts/MapChartD3/index.tsx

```diff
@@ -0,0 +1,552 @@
+    }, [mapData]);
+
+    const handleMouseEnter = (geo: MapChartFeature, e: MouseEvent): void => {
+        const [x, y] = pointer(e);
+        withActivity && setSelectedName(geo.properties.name);
+        withTooltip && setTooltipData({ top: y, left: x, content: geo.properties.name });
+    };
+
+    const handleMouseLeave = () => {
+        withActivity && setSelectedName(null);
+        withTooltip && setTooltipData({ top: 0, left: 0, content: null });
+    };
+
+    const handleMouseMove = (event: MouseEvent, context: CanvasRenderingContext2D, path: GeoPath) => {
+        const [x, y] = pointer(event);
+
+        // START Find hovered region
+        context.beginPath();
+        const generateContext = path.context(context);
+        const hoveredRegion = chartData.find((feature) => {
+            generateContext(feature as GeoPermissibleObjects);
+
+            return context.isPointInPath(
+                (x - transformRef.current.x) / transformRef.current.k,
+                (y - transformRef.current.y) / transformRef.current.k
+            );
+        });
+
+        // END Find hovered region
+
+        if (!hoveredRegion) {
+            if (!hoveredRegionRef.current) {
+                return;
+            }
+            hoveredRegionRef.current.properties.brightness = 0;
+            drawMap(context, path);
+            hoveredRegionRef.current = hoveredRegion;
+            handleMouseLeave();
+            return;
+        }
+
+        // START Highlight regions depends on brightness
+
+        if (hoveredRegion && hoveredRegion?.properties.value) {
+            if (hoveredRegionRef.current?.id === hoveredRegion.id) {
+                if (hoveredRegionRef.current.properties.brightness === hoveredRegion.properties.brightness) {
+                    handleMouseEnter(hoveredRegion, event);
+                    return;
+                }
+                handleMouseEnter(hoveredRegion, event);
+                hoveredRegion.properties.brightness = 0;
+
+                drawMap(context, path);
+            } else {
+                if (hoveredRegionRef.current) hoveredRegionRef.current.properties.brightness = 0;
+
+                hoveredRegion.properties.brightness = brightness;
+                drawMap(context, path);
+
+                handleMouseEnter(hoveredRegion, event);
+            }
+
+            hoveredRegionRef.current = hoveredRegion as MapChartFeature;
+            onPointOver(hoveredRegion);
+        } else if (
+            hoveredRegionRef.current &&
+            hoveredRegionRef.current.id !== hoveredRegion.id &&
+            hoveredRegionRef.current.properties.value
+        ) {
+            handleMouseLeave();
+            hoveredRegion.properties.brightness = 0;
+            hoveredRegionRef.current.properties.brightness = 0;
+            drawMap(context, path);
+            hoveredRegionRef.current = hoveredRegion;
+        }
+        // END Highlight regions depends on brightness
+    };
```

Comment:
you can write drawMap(context, path) right here and remove from all places

Reply:
we can not because we call `drawMap` method conditionally.